File: portfolio-server/src/main/java/com/maksudsharif/portfolioserver/model/Skill.java
package com.maksudsharif.portfolioserver.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.io.Serializable;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Skill implements Serializable
{

  private String name;
  private String years;

  @JsonInclude(JsonInclude.Include.NON_NULL)
  private String level;

  public Skill(String name)
  {
    this.name = name;
  }

  public static List<Skill> from(String... skills)
  {
    return skills != null && skills.length > 0 ? Arrays.stream(skills).map(Skill::new)
        .collect(Collectors.toList()) : Collections.emptyList();
  }
}

File: portfolio-client/src/app/components/model/skill.ts
export class Skill {


  constructor(name: string, level: string, years: string) {
    this.name = name;
    this.level = level;
    this.years = years;
  }

  name: string;
  level: string;
  years: string;

  static of(name: string, level: string, years: string): Skill {
    if (!level) {
      level = '0';
    }

    if (!name) {
      name = 'N/A';
    }

    if (!years) {
      years = '0';
    }

    return new Skill(name, level, years);
  }

  findLevel() {
    return this.level + '%';
  }
}
